app/views/planning_applications/validation/tasks/_application_details.html.erb

```diff
@@ -2,76 +2,75 @@
+        task_list.with_item(html_attributes: {id: "draw_red_line_boundary"}) do |item|
+          item.with_title(
+            text: "Draw red line boundary",
+            href: !(@planning_application.boundary_geojson.present? && @planning_application.officer_can_draw_boundary?) && planning_application_validation_sitemap_path(@planning_application)
           )
-        ) %>
-    <li id="constraints-validation-tasks" class="app-task-list__item">
-      <span class="app-task-list__task-name">
-        <%= govuk_link_to "Check constraints", planning_application_validation_constraints_path(@planning_application) %>
-      </span>
-      <div class="govuk-task-list__status app-task-list__task-tag">
-      <%= render(
-            StatusTags::ConstraintsComponent.new(
-              planning_application: @planning_application
-            )
-          ) %>
-      </div>
-    </li>
-    <% if @planning_application.validated? %>
-      <li id="check-description" class="app-task-list__item">
-        <span class="app-task-list__task-name">
-          Planning application has already been validated
-        </span>
-      </li>
-    <% else %>
-      <%= render(
-            TaskListItems::DescriptionChange.new(
-              planning_application: @planning_application
-            )
-          ) %>
-    <% end %>
-    <li id="development-type-for-reporting-task" class="app-task-list__item">
-      <span class="app-task-list__task-name">
-        <%= govuk_link_to "Add reporting details", edit_planning_application_validation_reporting_type_path(@planning_application) %>
-      </span>
+          item.with_status(text: render(StatusTags::DrawRedLineBoundaryComponent.new(boundary_geojson: @planning_application.boundary_geojson)))
+        end
 
-      <div class="govuk-task-list__status app-task-list__task-tag">
-      <%= render(
-            StatusTags::BaseComponent.new(
-              status: @planning_application.reporting_type_status
-            )
-          ) %>
-      </div>
-    </li>
-    <% if @planning_application.application_type.legislation_description %>
-      <% if @planning_application.validated? %>
-        <li id="check-legislative-requirements" class="app-task-list__item">
-          <span class="app-task-list__task-name">
-            Planning application has already been validated
-          </span>
-        </li>
-      <% else %>
-        <%= render(
-              TaskListItems::Validating::LegislationComponent.new(
-                planning_application: @planning_application
+        task_list.with_item(html_attributes: {id: "check-red-line-boundary"}) do |item|
+          task = Validation::CheckRedLineBoundaryTask.new(@planning_application)
+          item.with_title(text: task.task_list_link_text,
+            href: task.task_list_link)
+          item.with_status(text: render(StatusTags::BaseComponent.new(status: task.task_list_status)))
+        end
```

Comment:
I would have liked to move this block inside the Task class, so that you could pass in the task list component and have the task add itself as an item with all the relevant fields, but `render` isn't available from inside the task.

Reply:
I've addressed this for now by making it a partial.